### PR TITLE
Migrate katib to new test-infra

### LINKED ISSUE #1422: [Test] Migrate to new test-infra
**Description of your changes:**
Migrate kubeflow/katib to new test-infra, there's no difference between old test-infra and new test-infra from users' perspective.

Part of https://github.com/kubeflow/testing/issues/861

/cc @andreyvelich

### PR DESCRIPTION
**Pull request by PatrickXYS**
**What this PR does / why we need it**:
Migrate katib to new test-infra

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1422 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
NONE
```

/cc @andreyvelich 
/hold

### CODE REVIEW

**Comment by PatrickXYS** [comment]
```
 Failed: cannot unmarshall spec: cannot convert int64 to string
```

Checking

**Comment by PatrickXYS** [comment]
Same reason, roll back all the changes and see what's going on

**Comment by PatrickXYS** [comment]
```
INFO|2021-01-26T21:17:32|/src/kubeflow/testing/py/kubeflow/testing/argo_client.py|48| Parsing ApiException body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"workflows.argoproj.io \"kubeflow-katib-presubmit-e2e-v1beta1-1423-3f174af-3552-0e2b\" not found","reason":"NotFound","details":{"name":"kubeflow-katib-presubmit-e2e-v1beta1-1423-3f174af-3552-0e2b","group":"argoproj.io","kind":"workflows"},"code":404}
```

After applying workflow, it can't find corresponding workflow.

**Comment by PatrickXYS** [comment]
/test kubeflow-katib-presubmit

Kick off again to see if the issue happen

**Comment by PatrickXYS** [comment]
Same issue... Dig into it to see why it can't apply generate-workflow into remote argo cluster

**Comment by PatrickXYS** [comment]
It works this time, looks like those TTL and DDL configuration causes the issue

**Comment by PatrickXYS** [comment]
Removed ttl and ddl parameters since we have already configured default workflow spec:

```
configmap -o yaml
apiVersion: v1
data:
  workflowDefaults: |
    metadata:
      annotations:
        argo: workflows
    spec:
      ttlStrategy:
        secondsAfterSuccess: 604800
        secondsAfterCompletion: 604800
        secondsAfterFailure: 604800
      activeDeadlineSeconds: 86400
...
```
https://argoproj.github.io/argo/fields/#ttlstrategy

That's being said, WG does not need to configure ad-hoc activeDeadlineSeconds and ttlStrategy parameters.

/cc @andreyvelich

**Comment by PatrickXYS** [comment]
image

https://argo.kubeflow-testing.com/workflows/kubeflow-test-infra/kubeflow-katib-presubmit-e2e-v1beta1-1423-e99e074-4128-402f?tab=workflow&nodeId=kubeflow-katib-presubmit-e2e-v1beta1-1423-e99e074-4128-402f-1929247164

Works as expected

**Comment by andreyvelich** [comment]
> https://argoproj.github.io/argo/fields/#ttlstrategy

@PatrickXYS That makes sense.
Should we merge this ?

File: test/workflows/components/workflows-v1beta1.libsonnet [hunk 1]
```diff
@@ -179,7 +178,6 @@
         spec: {
           entrypoint: "e2e",
           // Cleanup workflow after 5 days.
```
Comment (andreyvelich) [inline test/workflows/components/workflows-v1beta1.libsonnet:180]: Can you remove this comment, please?

**Comment by PatrickXYS** [comment]
> > https://argoproj.github.io/argo/fields/#ttlstrategy
> 
> @PatrickXYS That makes sense.
> Should we merge this ?

It's already in, our new Argo cluster installed with v2.12.3 argo workflow

**Comment by PatrickXYS** [comment]
/unhold

@andreyvelich Can you help approve this PR and get it merged?

**Review by andreyvelich — APPROVED** [review]
/approve

**Comment by PatrickXYS** [comment]
Looks like you also need to lgtm @andreyvelich

**Pull request merged**